src/main/scala/PropertyChecker.scala

```diff
@@ -0,0 +1,408 @@
+
+      warnings :+ convertValueFacet(objectNode, "minInclusive", baseValue)
+      warnings :+ convertValueFacet(objectNode, "minExclusive", baseValue)
+      warnings :+ convertValueFacet(objectNode, "maxInclusive", baseValue)
+      warnings :+ convertValueFacet(objectNode, "maxExclusive", baseValue)
+
+      val minInclusive:Option[Either[Int, String]] = if (valueCopy.path("minInclusive").path("dateTime").isMissingNode) {
+        if (objectNode.path("minInclusive").isMissingNode) {
+          None
+        } else {
+          Option(Left(objectNode.get("minInclusive").asInt()))
+        }
+      } else {
+        if (objectNode.path("minInclusive").path("dateTime").isMissingNode) {
+          None
+        } else {
+          Option(Right(objectNode.get("minInclusive").get("dateTime").asText()))
+        }
+      }
+
+      val maxInclusive:Option[Either[Int, String]] = if (objectNode.path("maxInclusive").path("dateTime").isMissingNode) {
+        if (objectNode.path("maxInclusive").isMissingNode) {
+          None
+        } else {
+          Option(Left(objectNode.get("maxInclusive").asInt()))
+        }
+      } else {
+        if (objectNode.path("maxInclusive").path("dateTime").isMissingNode) {
+          None
+        } else {
+          Option(Right(objectNode.get("maxInclusive").get("dateTime").asText()))
+        }
+      }
+
+      val minExclusive:Option[Either[Int, String]] = if (objectNode.path("minExclusive").path("dateTime").isMissingNode) {
+        if (objectNode.path("minExclusive").isMissingNode) {
+          None
+        } else {
+          Option(Left(objectNode.get("minExclusive").asInt()))
+        }
+      } else {
+        if (objectNode.get("minExclusive").path("dateTime").isMissingNode) {
+          None
+        } else {
+          Option(Right(objectNode.get("minExclusive").get("dateTime").asText()))
+        }
+      }
+
+      val maxExclusive:Option[Either[Int, String]] = if (objectNode.path("maxExclusive").path("dateTime").isMissingNode) {
+        if (objectNode.path("maxExclusive").isMissingNode) {
+          None
+        } else {
+          Option(Left(objectNode.get("maxExclusive").asInt()))
+        }
+      } else {
+        if (objectNode.path("maxExclusive").path("dateTime").isMissingNode) {
+          None
+        } else {
+          Option(Right(objectNode.get("maxExclusive").get("dateTime").asText()))
+        }
+      }
+
+     (minInclusive, minExclusive, maxInclusive, maxExclusive) match {
+       case (Some(Left(minI)), Some(Left(minE)), _, _) => throw new MetadataError(s"datatype cannot specify both minimum/minInclusive ($minI) and minExclusive ($minE)")
+       case (Some(Right(minI)), Some(Right(minE)), _, _) => throw new MetadataError(s"datatype cannot specify both minimum/minInclusive ($minI) and minExclusive ($minE)")
+       case (_, _, Some(Left(maxI)), Some(Left(maxE))) => throw new MetadataError(s"datatype cannot specify both maximum/maxInclusive ($maxI) and maxExclusive ($maxE)")
```

Comment:
These two lines could be simplified to `Some(maxI), Some(maxE)`, it doesn't matter if they're left or right since you give the same error message for both.

Reply:
This is the case for some other matches here too.